null_resource.upload_user_flow (local-exec): <Error><Message>An error has occurred.</Message><ExceptionMessage>No MediaTypeFormatter is available to read an object of type 'HttpError' from content with media type 'application/problem+json'.</ExceptionMessage><ExceptionType>System.Net.Http.UnsupportedMediaTypeException</ExceptionType><StackTrace>   at System.Net.Http.HttpContentExtensions.ReadAsAsync[T](HttpContent content, Type type, IEnumerable`1 formatters, IFormatterLogger formatterLogger, CancellationToken cancellationToken)
null_resource.upload_user_flow (local-exec):    at Microsoft.Cpim.Web.Admin.ExceptionManager.RemoveStackTracesIfNeeded(HttpResponseMessage response) in C:\__w\1\s\src\Production\Web.Admin\ExceptionManager.cs:line 645
null_resource.upload_user_flow (local-exec):    at Microsoft.Cpim.Web.Admin.CpimRequestMessageHandler.&lt;SendAsync&gt;d__0.MoveNext() in C:\__w\1\s\src\Production\Web.Admin\CpimRequestMessageHandler.cs:line 40
null_resource.upload_user_flow (local-exec): --- End of stack trace from previous location where exception was thrown ---
null_resource.upload_user_flow (local-exec):    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
null_resource.upload_user_flow (local-exec):    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
null_resource.upload_user_flow (local-exec):    at System.Web.Http.HttpServer.&lt;SendAsync&gt;d__24.MoveNext()</StackTrace></Error>
null_resource.upload_user_flow: Creation complete after 1s [id=7675197542013355465]



data "http" "access_token" {
  url = "https://login.microsoftonline.com/${azurerm_aadb2c_directory.tenant.tenant_id}/oauth2/v2.0/token"
  
  request_headers = {
    "Content-Type" = "application/x-www-form-urlencoded"
  }
  
  request_body = "grant_type=client_credentials&client_id=${module.tenant.client_id}&client_secret=${module.tenant.client_secret}&scope=https://graph.microsoft.com/.default"
}

locals {
  access_token = jsondecode(data.http.access_token.response_body)["access_token"]
}

resource "null_resource" "upload_user_flow" {
  triggers = {
    always_run = timestamp()
  }

  provisioner "local-exec" {
    command = <<-EOT
      curl -X PUT \
           -H "Content-Type: application/json" \
           -H "Authorization: Bearer ${local.access_token}" \
           --data-binary "@userflow.json" \
           "https://main.b2cadmin.ext.azure.com/api/adminuserjourneys?tenantId=${var.rg_name}.onmicrosoft.com"
    EOT
  }
}



│ Error: Invalid index
│ 
│   on main.tf line 85, in locals:
│   85:   access_token = jsondecode(data.http.access_token.response_body)["access_token"]
│     ├────────────────
│     │ data.http.access_token.response_body is "{\"error\":\"invalid_resource\",\"error_description\":\"AADSTS500011: The resource principal named https://main.b2cadmin.ext.azure.com was not found in the tenant named azb2cpocemeka. This can happen if the application has not been installed by the administrator of the tenant or consented to by any user in the tenant. You might have sent your authentication request to the wrong tenant. Trace ID: 0d57c3ee-b077-4309-b733-5b74744f0800 Correlation ID: 46ce202e-27d6-4a69-b26d-a0f5eaddca86 Timestamp: 2024-03-04 23:14:41Z\",\"error_codes\":[500011],\"timestamp\":\"2024-03-04 23:14:41Z\",\"trace_id\":\"0d57c3ee-b077-4309-b733-5b74744f0800\",\"correlation_id\":\"46ce202e-27d6-4a69-b26d-a0f5eaddca86\",\"error_uri\":\"https://login.microsoftonline.com/error?code=500011\"}"
│ 
│ The given key does not identify an element in this collection value.
